### PR TITLE
Rasha

### PR DESCRIPTION
**Pull request by rasha-saber**

### CODE REVIEW

File: app/Http/Controllers/CarController.php [hunk 1]
```diff
@@ -27,72 +27,190 @@ public function index()
...
+            'image' => 'nullable|image|mimes:jpeg,png,jpg,gif,svg|max:2048',
+
         ]);
-        dd($request);
-        $data = [
-            'published' => isset($request->published),
+        $data['published'] = isset($request->published);
+        // التحقق من رفع الصورة
+        if ($request->hasFile('image')) {
+            $file = $request->file('image');
```
Comment (Nora-Lootfy) [inline app/Http/Controllers/CarController.php]: a little duplication here, don't worry we will work on that in the session

File: resources/views/car_details.blade.php [hunk 2]
```diff
@@ -26,16 +23,9 @@
...
-                alt="" class="card-img"
-                style="position: absolute; margin: auto; top: 50%; transform: translateY(-50%); width: 100%;height: 100%; object-fit: cover;" />
-           
-           
-             
-           
-           
-           
-              </div>
+              <img src="{{ asset('uploads/' . $car->image) }}" alt="" class="card-img" style="position: absolute; margin: auto; top: 50%; transform: translateY(-50%); width: 100%;height: 100%; object-fit: cover;" />
```
Comment (Nora-Lootfy) [inline resources/views/car_details.blade.php]: try to upload in one place, not uploads , assets/images and images this is confusing

**Review by Nora-Lootfy — APPROVED** [review]
Good job ya Rasha, i think you are coding just like a computer engineer.

Have a nice day😊

File: resources/views/car_details.blade.php (hunk 2 above)
Comment (rasha-saber) [reply resources/views/car_details.blade.php:26]: ok

File: app/Http/Controllers/CarController.php (hunk 1 above)
Comment (rasha-saber) [reply app/Http/Controllers/CarController.php]: اوك تمام ان شاء الله

File: resources/views/car_details.blade.php (hunk 2 above)
Comment (rasha-saber) [reply resources/views/car_details.blade.php]: حاضر

File: app/Http/Controllers/ProductController.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+{
+    use Common;
+    /**
+     * Display a listing of the resource.
+     */
+    public function indexR()
+    {
+
+
+        $products = Product::orderBy('created_at', 'desc')->take(3)->get();
```
Comment (Nora-Lootfy) [inline app/Http/Controllers/ProductController.php:19]: good job ya Rasha,,

however,, there is a simpler function called `latest()` do the same as `orderBy('created_at', 'desc')` but your code still okay

File: app/Http/Controllers/ProductController.php [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    /**
+     * Store a newly created resource in storage.
+     */
+    public function store(Request $request)
+    {
+        $data = $request->validate([
+            'title' => 'required|string|max:255',
+            'short_description' => 'required|string|max:1000',
+            'price' => 'required|numeric',
+            'image' => 'nullable|image|mimes:jpeg,png,jpg,gif,svg|max:2048',
```
Comment (Nora-Lootfy) [inline app/Http/Controllers/ProductController.php:40]: why image is `nullable` here, all products must have image

we make it nullable in update as we can update it or not keep the old image, but in create we have to provide an image

File: app/Models/Product.php [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\SoftDeletes;
+
+class Product extends Model
+{
+    use HasFactory, softDeletes;
```
Comment (Nora-Lootfy) [inline app/Models/Product.php:11]: why you repeat `HasFactory`, also s in `SoftDeletes` is capital

**Review by Nora-Lootfy — APPROVED** [review]
Good job ya Rasha

**Comment by Nora-Lootfy** [comment]
please ya Rasha,, remember to close your pull request after giving you the review and reopen a new one after submitting a new task

**Pull request closed without merging**